### ISSUE TITLE
Device dropdown can open the wrong waveIn input device (CBS_SORT index mismatch)

### ISSUE THREAD

**Issue by sibiryakov** [sibiryakov/wow-and-flutter-analyzer#4]
## Summary

When two or more audio input devices are present, selecting a device from the input dropdown can cause the app to actually open a *different* physical device for recording, with no error shown.

## Root cause

`CWfguiDlg::FillDevices()` (`wfguiDlg.cpp:540-565`) enumerates devices via `waveInGetNumDevs()`/`waveInGetDevCaps()` and adds each device name to the `IDC_DEVICES` combo box in enumeration order:

```cpp
nDevices = waveInGetNumDevs();
for (nC1 = 0; nC1 < nDevices; ++nC1) {
    waveInGetDevCaps(nC1, &stWIC, sizeof(WAVEINCAPS));
    pBox->AddString(stWIC.szPname);
}
```

However, the combo box is declared with the `CBS_SORT` style (`wfgui.rc:111`), so Windows silently **re-sorts the entries alphabetically** as they're added. The combo box's selection *index* therefore no longer corresponds to the original `waveIn` device ID used during enumeration.

Later, `CWfguiDlg::OpenDevice()` (`wfguiDlg.cpp:610-642`) passes the combo box's current index straight through as the numeric waveIn device ID:

```cpp
waveInOpen(&m_hWaveIn, pDevices->GetCurSel(), &m_stWFEX, ...);
```

If device names don't happen to sort in the same order they were enumerated by the OS, this opens the wrong physical device.

## Secondary issue (same area)

`ON_CBN_SELCHANGE(IDC_DEVICES, OnCbnSelchangeDevices)` is not wired into the message map (`BEGIN_MESSAGE_MAP(CWfguiDlg, CDialog)`, `wfguiDlg.cpp:155-171`). `OnCbnSelchangeDevices()` — which checks whether the device supports the required format — currently only runs once at startup for the default (index 0) selection, via a direct call inside `FillDevices()` (`wfguiDlg.cpp:562`). Manually changing the dropdown never re-runs this validation.

## Repro

1. Have 2+ audio input devices attached whose names do not alphabetize in the same order they're enumerated by Windows (e.g. a USB interface enumerated as device 0 by the OS, but named alphabetically after the built-in mic).
2. Launch the app, select the device you expect from the dropdown.
3. Click Start.
4. Observe recording is actually captured from a different device than the one selected in the UI.

## Suggested fix

Store the actual waveIn device ID alongside each combo box entry (e.g. via `CComboBox::SetItemData`) instead of relying on the sorted display index, and use that stored ID in `waveInOpen`. Also wire up `ON_CBN_SELCHANGE` so format validation re-runs whenever the user changes the selection.

## Location references
- `wfguiDlg.cpp:540-565` — `FillDevices()`
- `wfguiDlg.cpp:567-585` — `OnCbnSelchangeDevices()`
- `wfguiDlg.cpp:610-642` — `OpenDevice()`
- `wfguiDlg.cpp:155-171` — message map (missing `ON_CBN_SELCHANGE`)
- `wfgui.rc:111` — `IDC_DEVICES` combo box (`CBS_SORT`)
